.github/workflows/realm-generator.yml

```diff
@@ -0,0 +1,64 @@
+name: Realm Generator
+
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - 'generator/**'
+      - .github/workflows/realm-generator.yml
+
+  pull_request:
+    paths:
+      - 'generator/**'
+      - .github/workflows/realm-generator.yml
+
+jobs:
+  Generator_CI:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest] # windows is not tested ue to bug in how error messages are rendered
```

Comment:
I suggest we test on linux only. It works the same on all platforms and fails the same as well.

Reply:
As you can see from the comment on windows, that is actually not guaranteed. This should run rarely, so I think it is okay to test both platform.. I'll raise a bug report regarding the windows issue

Reply:
Is this a bug in Actions or Windows in general

Reply:
This is bug in error output on windows... it still works, it just doesn't look as good as on the other platforms

Reply:
I see, ok lets leave it as a matrix test